enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,568 @@
+
+The node will initially boot with the network configuration specified in
+`preprovisioningNetworkDataName` and then have the configuration in the
+machine-config applied.
+
+Only the first step of the process is new for this feature. The rest is
+standard baremetal scaleout when using static IPs.
+
+### Risks and Mitigations
+
+Because this gives the deployer a great deal of control over the bridge
+configuration for OVNK, they will have the ability to configure it incorrectly,
+which may cause problems either on initial deployment or at any point after.
+We will want to ensure that must-gather is collecting enough information about
+the bridge configuration for us to determine whether it is compliant with our
+guidelines, and we must have clear instructions on what is required for br-ex.
+
+Note, however, that if an incorrect bridge configuration is applied NMState
+will usually catch that and roll back the changes.
+
+### Drawbacks
+
+OpenShift has historically avoided per-host configuration for cluster nodes.
+This is a large departure from that philosophy, but that is unavoidable.
+Certain aspects of networking (notably static IPs) are by nature host-specific
+and cannot be handled any other way.
+
+Using a machine-config to do initial deployment and a
+NodeNetworkConfigurationPolicy to make day 2 changes will require a manual sync
+of the NMState configuration. Unfortunately this is unavoidable because the
+Kubernetes-NMState operator is not part of the core payload so it (and its
+CRDs) cannot be used for initial deployment. A possible future improvement
+in this area would be to have Kubernetes-NMState "adopt" the configuration
+applied by this process and generate an NNCP automatically when it is
+installed.
+
+
+## Design Details
+
+### Open Questions [optional]
+
+* There has been resistance to using the NMState API in OpenShift in the past
+  because it is not compliant with the OpenShift API Guidelines. There is work
+  underway in NMState to address this, but is that mandatory for this to be
+  implemented?
+
+  Previous discussions:
+  https://github.com/openshift/enhancements/pull/1267#discussion_r1013320148
+
+  Some initial work to address this concern:
+  https://github.com/nmstate/nmstate/pull/2338
+
+* If changes are made to the bridge configuration on day 2, will OVNKubernetes
+  handle those correctly without a reboot? If not, how do we orchestrate a
+  reboot?
+
+  Reply from @cgoncalves:
+  > OVN-Kubernetes will not handle the replacement of the gateway interface (enp2s0 in the example above). This is an existing limitation where OVN-Kubernetes installs an OpenFlow rule in br-ex that states the gateway interface is the OVS port ID of enp2s0 (port 1) and where attaching a new interface, say enp3s0, its OVS port ID will be different hence no egress traffic forwarding.
+
+  That doesn't have to block this feature being implemented, but it's something
+  we should look into as a followup with the OVNK team.
```

Comment:
In the MTU case, is configure-ovs responsible for changing the MTU or does it just pick up the changed MTU of the underlying interface? I was thinking that you had to manually modify the interface as part of the process, which you would just do with Kubernetes-NMState here.

It would be good if we can come up with a list of day 2 features that might require bridge changes so we know what use cases to target.

Reply:
It just picks up the changed MTU of the underlying interface that a user is responsible for configuring correctly.

> which you would just do with Kubernetes-NMState here

If you mean changing br-ex and children interfaces, then we need to make sure this works with Kubernetes-NMState. That's what I meant that needs to be considered within the scope of the enhancement.

I am aware of any other features that depend directly on changing network configuration on day 2 (perhaps this could also help when enabling dualstack) but I do know some other features depend on the MTU migration itself, like enabling ipsec or sdn<->ovn migration.